### PR TITLE
Upgrade to Bootstrap 5 for Angular

### PR DESCRIPTION
**Pull request by mraible**
Upgrading to Bootstrap 5 will likely be a lot of work because of the downstream library dependencies. However, I figured I'd start this branch so others can contribute, and we can put a bug bounty on it.

See https://github.com/ng-bootstrap/ng-bootstrap/issues/3899
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by deepu105** [comment]
I hanged the title to be Angular specific as updating to BS for all 3 framework is huge work and IMO should be 3 separate tickets with 500$ each :wink:

**Comment by DanielFran** [comment]
@mraible @mshima [v12](https://github.com/ng-bootstrap/ng-bootstrap/releases/tag/12.0.0) has been released

**Comment by mshima** [comment]
> @mraible @mshima [v12](https://github.com/ng-bootstrap/ng-bootstrap/releases/tag/12.0.0) has been released

👍

**Comment by mraible** [comment]
I updated my Angular Mini-Book to use v12 yesterday. Here are the dependencies it adds when you use `ng add @ng-bootstrap/ng-bootstrap`:

```json
"@ng-bootstrap/ng-bootstrap": "^12.0.0",
"@popperjs/core": "^2.10.2",
"bootstrap": "^5.1.3",
```

**Comment by mraible** [comment]
@mshima Anything I can do to help with this one?

**Comment by mshima** [comment]
Not right now, I have a few changes that needs to be pushed.
I need to do some more testing.
And implement rtl support.

**Comment by mshima** [comment]
@mraible should be ready now.
But not sure if v7.7.0 should include this, I think angular is broken at v7.6.0 due to transitive dependency change.

File: generators/client/templates/angular/src/main/webapp/content/scss/global.scss.ejs [hunk 1]
```diff
@@ -224,4 +228,29 @@ ui bootstrap tweaks
   white-space: nowrap;
 }
 
+.dropdown-menu {
+  padding-left: 0px;
+}
+
+/* ==========================================================================
+postcss-rtl currently have problem with inline css
```
Comment (mshima) [inline generators/client/templates/angular/src/main/webapp/content/scss/global.scss.ejs:236]:
```suggestion
angular-cli removes postcss-rtl processed inline css
```

File: generators/client/templates/angular/webpack/webpack.custom.js.ejs [hunk 2]
```diff
@@ -115,6 +118,21 @@ module.exports = async (config, options, targetOptions) => {
...
+  const uses = scssRule.rules.flatMap(r => r.use || r.oneOf.flatMap(o => o.use));
+  const postcssLoaderOptions = uses.filter(u => u.loader.includes('postcss-loader')).map(u => u.options);
+  postcssLoaderOptions.forEach(options => {
+    const generateOptions = options.postcssOptions;
+    options.postcssOptions = (loader) => {
+      const postcssOptions = generateOptions(loader);
+      postcssOptions.plugins.push(postcssRTLCSS());
+      return postcssOptions;
+    };
+    options.postcssOptions.config = false;
```
Comment (mshima) [inline generators/client/templates/angular/webpack/webpack.custom.js.ejs:132]:
```suggestion
    // https://github.com/angular/angular-cli/blob/4b9199d97f3bcb10710b5049b04d8a04436e2dd4/packages/angular_devkit/build_angular/src/webpack/configs/styles.ts#L201-L202
    options.postcssOptions.config = false;
```

**Comment by mshima** [comment]
Failures are not related with this PR.

File: generators/client/templates/angular/package.json [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
     "@angular/common": "13.2.2",
-    "bootstrap": "4.6.1",
     "@fortawesome/angular-fontawesome": "0.10.1",
     "@fortawesome/fontawesome-svg-core": "1.3.0",
     "@fortawesome/free-solid-svg-icons": "6.0.0",
-    "@ng-bootstrap/ng-bootstrap": "11.0.0",
+    "@ng-bootstrap/ng-bootstrap": "12.0.0",
     "@ngx-translate/core": "14.0.0",
     "@ngx-translate/http-loader": "7.0.0",
+    "@popperjs/core": "^2.10.2",
```
Comment (DanielFran) [inline generators/client/templates/angular/package.json:10]:
```suggestion
    "@popperjs/core": "^2.10.2",
```
```suggestion
    "@popperjs/core": "2.10.2",
```

Comment (mraible) [reply generators/client/templates/angular/package.json:10]: Even better: go for the latest `2.11.2`

Comment (mshima) [reply generators/client/templates/angular/package.json:10]:
```suggestion
    "@popperjs/core": "2.11.2",
```

**Comment by DanielFran** [comment]
@mshima can you update based on main branch, please?

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@mraible @mshima Can we merge it?

**Comment by mshima** [comment]
@DanielFran i think we can go ahead

**Pull request merged**